CHANGELOG.md

```diff
@@ -6,6 +6,7 @@
 - Added ability to encode ProjectSpec to JSON [#545](https://github.com/yonaskolb/XcodeGen/pull/545) @ryohey
 - Added ability to skip tests [#582](https://github.com/yonaskolb/XcodeGen/pull/582) @kadarandras
 - Added ability to set `attributes` on build files [#583](https://github.com/yonaskolb/XcodeGen/pull/583) @min
+- Allow using environment variables in the form of `${SOME_VARIABLE}`. This might be a **breaking** change when a target template attribute is also defined as an environment variable [#594](https://github.com/yonaskolb/XcodeGen/pull/594) @tomquist
```

Comment:
Would this actually be a breaking change? It seems the variables are replaced AFTER template attributes, so they wouldn’t interfere. Thinking about this now, could you add a test that shows who wins, in this case the template attributes.

Reply:
Unfortunately it's the other way round since variable expansion is done on the JSON dictionary and template attributes get expanded later when the JSON dictionary is converted into a `Project` value.

I added a spec that documents this behavior. 

In general I think this order is good because it allows greatest flexibility when using variables. E.g. you could define which template to use via environment variable which would not be possible if variables were replaced after template attributes.

We could also think about using a different placeholder scheme for variables, e.g. `$(SOME_VARIABLE)`. Not sure if it is a good idea to have two different placeholder types though.